pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/GetDsmDrugSuggestionsRoute.java

```diff
@@ -65,11 +66,16 @@ private DrugSuggestionResponse getDrugSuggestions(String drugQuery, int limit) {
 
         // now rank the matches in a way that puts left-most matches near the top,
         // favoring word start matches
+        var suggestionComparator = new StringSuggestionTypeaheadComparator(upperDrugQuery);
         List<DrugSuggestion> sortedSuggestions = new ArrayList<>();
-        drugMatches.stream().sorted(new DrugTypeaheadComparator(drugQuery)).limit(limit).forEach(drug -> {
-            int offset = drug.getName().toUpperCase().indexOf(upperDrugQuery);
-            sortedSuggestions.add(new DrugSuggestion(drug, Collections.singletonList(new PatternMatch(offset, drugQuery.length()))));
-        });
+        drugMatches.stream()
+                .sorted((lhs, rhs) -> suggestionComparator.compare(lhs.getName(), rhs.getName()))
+                .limit(limit)
+                .forEach(drug -> {
```

Comment:
I believe that if you convert the forEach to `map` you won't need to do an `add` to a blank list. You can just follow the `map` in pipeline with a collect(toList()). More compact and no mutation of objects (in this case we are mutating the list).